### PR TITLE
WIP: Add new 'powerShell/getCommandInformation' message

### PR DESCRIPTION
**Pull request by corbob**
This adds a new getCommandInformation message used by the [Command Explorer](https://github.com/PowerShell/vscode-powershell/pull/1972) to get the details of a given command.

Pre-omnisharp we were optionally sending the name of a command in the `powerShell/getCommand` message, and had code to get information for a single command in that [instance](https://github.com/PowerShell/PowerShellEditorServices/blob/legacy/1.x/src/PowerShellEditorServices.Protocol/Server/LanguageServer.cs#L552-L555). It appears this was removed.I am split on if it's better being separate handlers. On the one hand, it seems it might be easier to understand being separate handlers, but it adds code duplication that might not be needed.

If kept as separate messages as shown here, there is some code cleanup we can do in the getCommand message handler as it's currently sending back values that are guaranteed to be empty and can be removed.

### CODE REVIEW

File: src/PowerShellEditorServices/Services/PowerShellContext/Handlers/GetCommandInformationHandler.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        public string Name { get; set; }
+        public string ModuleName { get; set; }
+        public string DefaultParameterSet { get; set; }
+        public Dictionary<string, ParameterMetadata> Parameters { get; set; }
+        public System.Collections.ObjectModel.ReadOnlyCollection<CommandParameterSetInfo> ParameterSets { get; set; }
+    }
+
+    public class GetCommandInformationHandler : IGetCommandInformationHandler
+    {
+        private readonly ILogger<GetCommandInformationHandler> _logger;
```
Comment (TylerLeonhardt) [inline src/PowerShellEditorServices/Services/PowerShellContext/Handlers/GetCommandInformationHandler.cs]: Codacy Issue found: [Remove this unread private field '_logger' or refactor the code to use its value.](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4980809)

**Comment by TylerLeonhardt** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- src/PowerShellEditorServices/Services/PowerShellContext/Handlers/GetCommandInformationHandler.cs  3
         

Clones added
============
- src/PowerShellEditorServices/Services/PowerShellContext/Handlers/GetCommandInformationHandler.cs  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4980809&bid=16449950)

**Comment by TylerLeonhardt** [comment]
I think I'd want them in the same handler. Since they both return the same data. It'd be a smaller change too :)

**Comment by TylerLeonhardt** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/PowerShellEditorServices/Services/PowerShellContext/Handlers/GetCommandHandler.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4980809&bid=16449950)

File: src/PowerShellEditorServices/Services/PowerShellContext/Handlers/GetCommandHandler.cs [hunk 2]
```diff
@@ -17,7 +17,10 @@ namespace Microsoft.PowerShell.EditorServices.Handlers
     [Serial, Method("powerShell/getCommand")]
```
Comment (BoryaBorya) [inline src/PowerShellEditorServices/Services/PowerShellContext/Handlers/GetCommandHandler.cs:17]:
```suggestion
    [Serial, Method("powerShell/getCommand")]
```

**Comment by corbob** [comment]
Copy/Pasta'd from the vscode-powershell one, it appears this PR's only 1 year old 🤷‍♂️ 

Given that we're approaching 2 years since I started this PR and I don't foresee myself having the time to work on it anytime soon, I'll close these PRs.

I'll keep my branches around in case anyone wants to pick up the work and use them as a starting place (be forewarned: there are multiple merge conflicts you'll have to resolve to bring it up to date).

**Pull request closed without merging**